### PR TITLE
Sbml improvements - notes, gene IDs, group IDs, and test

### PR DESCRIPTION
**Pull request by akaviaLab**
As requested in #885, the commits are atomic.
This pull request covers multiple SBML improvements
1) Removes parenthesis, unless their removal results in a parsing error #887 
2) Parses SBML notes of various formats #888 
3) Converts gene, reaction, metabolite ID back and forth, including dealing with non-ascii characters that are not allowed in SBML IDs, should tackle #883 and has some progress towards https://github.com/opencobra/schema/issues/6 and https://github.com/opencobra/schema/issues/8
4) Added conversion of group IDs. Currently this uses the prefix G_ but it can be changed. One option is P_ for grouP
5) Added test of notes and GPR

### CODE REVIEW

File: cobra/test/test_io/test_sbml.py [hunk 1]
```diff
@@ -388,3 +388,62 @@ def test_identifiers_annotation():
...
+                            '41-21)6-39-44(36,37)42-43(34,35)38-5-10-13(29)15(31)20(40-10)27-3-1-2-9(4-27)18('
+                            '23)33/h1,3-4,7-8,10-11,13-16,20-21,29-32H,2,5-6H2,(H2,23,33)(H,34,35)(H,36,37)(H2,22,24,'
+                            '25)/p-2/t10-,11-,13-,14-,15-,16-,20-,21-/m1/s1',
+                   'chebi': 'CHEBI:57945'}
+    }
+    rxnNotesValue = {'CONFIDENCE_LEVEL': '4', 'NOTES': 'NCD', 'SUBSYSTEM': 'Propanoate metabolism',
+                     'GENE_ASSOCIATION': '(HGNC:8546 and HGNC:8548) or (HGNC:8547 and HGNC:8548)'}
+    rxnAnnotValue = {'sbo': 'SBO:0000176', 'ec-code': '1.1.1.27', 'pubmed': ['10108', '21765']}
+
+    for metID in metNoteValues.keys():
```
Comment (Midnighter) [inline cobra/test/test_io/test_sbml.py]: You don't need the `.keys()` method. A `for` loop automatically iterates over the keys in a dictionary.

```suggestion
    for met_id in metNoteValues:
```

**Comment by Midnighter** [comment]
If you like I can fix these last little issues. I feel bad making you work so much on style problems when we should be using a tool like [black](https://pypi.org/project/black/).

**Comment by akaviaLab** [comment]
Current status - I can't figure out why 3.5 is complaining on import order. Suggestions would be appreciated. Can we just ignore this? You can also remove the kludge fix (6ec605b0dc27f38b73b932a962c010a0845fc0e3), since it didn't fix 3.5 complaining.

2.7 fails. I can't manage to run 2.7 tests on my system. I'm getting an error importing sympy.

If you can fix 3.5 and 2.7, I would appreciate it.

**Comment by cdiener** [comment]
I think 3.5 is complaining about import order since isort is only run there (doesn't make sense to re-run style checks on all versions).

**Comment by Midnighter** [comment]
Looks really good now. Thank you for the fixes and your persistence!

**Comment by akaviaLab** [comment]
Question - how do these functions relate to cobra.manipulation.escape_ID(model)?
Should I have modified escape_ID instead of modifying it in place? Is escape_ID obselete and shoudl be erased?

**Pull request merged**